src/components/views/calendar/Calendar.tsx

```diff
@@ -1,145 +1,281 @@
-//     //     IsAllDay: isAllDay,
-//     //     Description: description
-//     //   }]
-//     // }
-//     let newEvent: any = {
-//       dataSource: [{
-//         Subject: subject,
-//         EndTime: new Date(`${endTime}`),
-//         StartTime: new Date(`${startTime}`),
-//         IsAllDay: isAllDay,
-//         Description: description
-//       }]
-//     }
-//     setEvents(newEvent)
-//     // console.log(scheduleObj.current?.eventsProcessed) //NOT undefined
-//   }, [scheduleObj])
-//   console.log('events', events)
-//   console.log(scheduleObj)
+    };
 
+    fetchData();
+  }, []);
+
+  const dataManager = new DataManager({
+    url: `https://rr-users-calendars-service-3e13398e3ea5.herokuapp.com/api/v1/users/${user.data.id}/calendar_events`,
+    adaptor: new WebApiAdaptor(),
+    crossDomain: true,
+    headers: [
+      { Authorization: `${currentToken}` },
+      { "Content-Type": "application/json" },
+    ],
+  });
+
+  const save_icon = "e-save-icon e-icons";
+  const save_button =
+    "e-schedule-dialog e-control e-btn e-lib e-primary e-event-save e-flat";
+
+  // For future API call to delete CalendarEvent in closePopup
+  const delete_event = "e-btn-icon e-icons e-delete-icon";
+
+  const closePopup = (args: PopupCloseEventArgs) => {
+    console.log("close Popup Here");
+    if (args.event && args.event.target) {
+      const target = args.event.target as HTMLElement;
+      if (target.className === save_icon || target.className === save_button) {
+        const newEvent: ScheduleEvent = {
+          Id: scheduleData.length + 1,
+          Subject: (args.data as any).Subject,
+          Description: (args.data as any).Description,
+          StartTime: new Date((args.data as any).StartTime),
+          EndTime: new Date((args.data as any).EndTime),
+        };
+        const apiFormattedEvent = transformToApiFormat(newEvent, user.data.id);
+        dataManager.insert(apiFormattedEvent);
+      }
+    }
+  };
+
+  const dragStopEvent = (args: DragEventArgs) => {
+    const newEvent: ScheduleEvent = {
+      Id: scheduleData.length + 1,
+      Subject: args.data.Subject,
+      Description: args.data.Description,
+      StartTime: new Date(args.data.StartTime),
+      EndTime: new Date(args.data.EndTime),
+    };
+    const apiFormattedEvent = transformToApiFormat(newEvent, user.data.id);
+    dataManager.insert(apiFormattedEvent);
+  }
 
   return (
     <>
       {user.data.id ? (
         <>
-          <ScheduleComponent eventSettings={newNewEvent} ref={scheduleObj}>
-            <Inject services={[Day, Week, Month, Agenda]} />
-          </ScheduleComponent>
+          {windowLocation.includes("calendar") ? (
+            <ScheduleComponent
+              eventSettings={{ dataSource: scheduleData }}
+              ref={scheduleObj}
+              popupClose={closePopup}
+              allowSwiping={true}
+              allowDragAndDrop={true}
+              dragStop={dragStopEvent}
```

Comment:
While I fixed one duplication problem I created another. This can be fixed once a destroy option is provided